Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -0,0 +1,91 @@
+import mongoc
+
+/// Options to use when listing available databases.
+internal struct ListDatabasesOptions: Encodable {
+    /// Optional `Document` specifying a filter that the listed databases must pass.
+    internal var filter: Document?
+
+    /// Optionally indicate whether only names should be returned.
+    /// This is internal and only used for implementation purposes. Users should use `listDatabaseNames` if they want
+    /// only names.
+    internal var nameOnly: Bool?
+
+    /// Convenience constructor for basic construction
+    internal init(filter: Document?, nameOnly: Bool?) {
+        self.filter = filter
+        self.nameOnly = nameOnly
+    }
+}
+
+/// A struct modeling the information returned from the `listDatabases` command about a single database.
+public struct DatabaseSpecification: Codable {
+    /// The name of the database.
+    public let name: String
+
+    /// The amount of disk space consumed by this database.
+    public let sizeOnDisk: Int
+
+    /// Whether or not this database is empty.
+    public let empty: Bool
+
+    /// For sharded clusters, this field includes a document which maps each shard to the size in bytes of the database
+    /// on disk on that shard. For non sharded environments, this field is nil.
+    public let shards: Document?
```

Comment:
Accidentally omitted this field before. Because the structure of this document is based on the sharded topology, we can't use a `Codable` struct here in place of a `Document`.